akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -155,6 +155,7 @@ private[akka] object FanIn {
       if (input.inputsDepleted) {
         if (marked(id)) markedDepleted += 1
         depleted(id) = true
+        onDepleted(id)
```

Comment:
Yes, I came to the same conclusion. The only place that uses this is FlexiMerge, which built upon the misbehavior, but I’ll fix it.